cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -833,10 +834,9 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
     Set<EntityScope> entityScopes = searchRequest.getEntityScopes();
     List<SearchTerm> searchTerms = new LinkedList<>();
     Consumer<String> termAdder = determineSearchFields(namespace, entityScopes, searchTerms);
-    String searchQuery = searchRequest.getQuery();
-    for (String term : Splitter.on(SPACE_SEPARATOR_PATTERN).omitEmptyStrings().trimResults().split(searchQuery)) {
-      termAdder.accept(term);
-    }
+
+    QueryParser.parse(searchRequest.getQuery()).forEach(queryTerm -> termAdder.accept(queryTerm.getTerm()));
```

Comment:
A nicer way to write it would be:

```
QueryParser.parse(searchRequest.getQuery()).stream()
  .map(QueryTerm::getTerm)
  .forEach(termAdder);
```

Reply:
fixed